AirSkyBoat/AirSkyBoat
# Fixes Aqueous Orichalcum Ingot Synth, Triton Ring and Poseidon Ring Vendor Price, and Poseidon Ring Effect

This was a multi-issue bug report.  Proof/Documentation was included in the 'Steps to Check these Changes'



**_I affirm:_**


## Please enter a player-facing description
- Added Key Item requirement to Aqueous Orichalcum Ingot
- Updated vendor prices for Orichalcum Ring, Triton Ring, and Poseidon's Ring
- Fixed Poseidon's Ring on-use effect.

## What does this pull request do? (Please be technical)


Closes https://github.com/HorizonFFXI/HorizonXI-Issues/issues/1697

## Steps to test these changes


1 - Aqueous Orichalcum Ingot Recipe is not requiring the Gold Ensorcellment
[2006 Era Wiki Entry](https://ffxiclopedia.fandom.com/wiki/Aqueous_Orichalcum?oldid=91366)
[Somepage DB Entry](http://ffxi.somepage.com/itemdb/5840)
 - `!additem 4096`
 - `!additem 738`
 - `!additem 739 3`
 - `!additem 1645`
 - `!additem 1646`
 - `!additem 1648`
 - Attempt to synth the Aqueous Orichalcum Ingot (It will fail now)
 - `!addkeyitem 2001`
 - Attempt to synth the Aqueous Orichalcum Ingot (It will succeed)

2 - The vendor price of Poseidon's Ring is wrong (4069 vs 45795)
[Wiki Price listed at 45795](https://ffxiclopedia.fandom.com/wiki/Poseidon%27s_Ring)
[JP Wiki Price listed at 45795](https://wiki-ffo-jp.translate.goog/html/9155.html?_x_tr_sl=es&_x_tr_tl=en&_x_tr_hl=en&_x_tr_pto=wapp)
 - `!additem 14656`
 - Sell the item to a vendor to see updated price

3 - The vendor price Orichalcum Ring HQ (Triton Ring) is wrong (4000 vs 37079)
[JP Wiki Price listed at 31450](https://wiki-ffo-jp.translate.goog/html/2177.html?_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=en&_x_tr_pto=wapp)
 - `!additem 14616`
 - Sell the item to a vendor to see updated price

4 - Poseidon's Ring is unusable
[JP Wiki Explaining AI](https://wiki-ffo-jp.translate.goog/html/9155.html?_x_tr_sl=es&_x_tr_tl=en&_x_tr_hl=en&_x_tr_pto=wapp)
 - `!additem 14656 2`
 - Equip both rings
 - `!zone 101`
 - Use the Poseidon's Ring to summon a Water Elemental Pet
 - Attempt to use a second ring while the first pet is active (Will receive error message to prevent multiple pets)
 - Perform an action on any mob to engage the pet
 - Zone into Sandy and attempt to use the ring (Will receive error message to prevent pets from non-pet zones)

## Special Deployment Considerations



- Updated `petutils.cpp` to perform a basic check for the item (Checks if player has the item) and that the pet being summoned is the Water Spirit ID.  If it is, then the pet is allowed to set its level to the same as player.  This was done to prevent `[debug]` message showing that a non-smn player was using an avatar and setting the level of the pet to 1.
- This is not full-proof as the functionality of `charutils::HasItem()` is only checking the inventory of the player -- However, the edge-case associated with having this item and casting this specific pet might be enough as it appears to be the only item in the game that allows a non-smn to summon a Water Spirit.
- Currently the Avatar Perpetuation Cost for the Water Spirit from this ring is using JP Wiki values for the stated level of 16MP/3sec at Lv65.
  - At Lv75, the perpetuation cost increases to 18MP/3sec at Lv75 -- It might be worth digging into this deeper to check if this was a fixed perpetuation cost or not.  NA Wiki states that the perpetuation cost is only 15/MP Tick at 75, but I am more inclined to believe JP Wiki and that the perpetuation cost increases with the level of the player/pet.
